helm/cluster-aws/files/opt/bin/kubelet-aws-config.sh

```
@@ -531,6 +529,17 @@ z1d.xlarge) max_pods=$(((4-$reserved_eni)*(15-1))) ;;
 	;;
 esac
 
+{{- end }}
+
+# max pods can't be greater than the number of available IPs in nodeCidrMaskSize
+node_cidr_mask_size="{{ .Values.global.connectivity.network.pods.nodeCidrMaskSize }}"
+if [[ -n "$node_cidr_mask_size" ]]; then
```

It's always a non-empty integer as of the schema, I think. Let's rather fail if empty or invalid?